### PR TITLE
e2e: Refactored service helpers to be consistent with new API.

### PR DESCRIPTION
**Pull request by bwplotka**
* Simplified and consistent e2e usage
* Removed unnecessary rule test

### CODE REVIEW

**Comment by yeya24** [comment]
Looks good overall. I am good if CI pass.

File: test/e2e/e2ethanos/services.go [hunk 1]
```diff
@@ -813,56 +702,56 @@ http {
...
 	image := "minio/minio:RELEASE.2019-12-30T05-45-39Z"
 	minioKESGithubContent := "https://raw.githubusercontent.com/minio/kes/master"
 	commands := []string{
 		"curl -sSL --tlsv1.2 -O '%s/root.key'	-O '%s/root.cert'",
 		"mkdir -p /data/%s && minio server --certs-dir /shared/data/certs --address :%v --quiet /data",
 	}
 
-	if err := os.MkdirAll(filepath.Join(env.SharedDir(), "data", "certs", "CAs"), 0750); err != nil {
-		return nil, errors.Wrap(err, "create certs dir")
+	f := e2e.NewInstrumentedRunnable(e, fmt.Sprintf("minio-", name)).
```
Comment (yeya24) [inline test/e2e/e2ethanos/services.go:737]: This needs %s or `+`

File: test/e2e/e2ethanos/services.go [hunk 2]
```diff
@@ -813,56 +702,56 @@ http {
...
 	}
 
-	if err := ioutil.WriteFile(filepath.Join(dir, "nginx.conf"), []byte(conf), 0600); err != nil {
-		return nil, errors.Wrap(err, "creating nginx config file failed")
+	if err := ioutil.WriteFile(filepath.Join(f.Dir(), "nginx.conf"), []byte(conf), 0600); err != nil {
+		return e2e.NewErrInstrumentedRunnable(name, errors.Wrap(err, "creating nginx config file failed"))
 	}
 
-	nginx := e2e.NewInstrumentedRunnable(e, fmt.Sprintf("nginx-%s", name)).WithPorts(map[string]int{"http": 80}, "http").Init(
+	return e2e.NewInstrumentedRunnable(e, fmt.Sprintf("nginx-%s", name)).WithPorts(map[string]int{"http": 80}, "http").Init(
```
Comment (matej-g) [inline test/e2e/e2ethanos/services.go:717]: I think this is causing the CI to fail, I guess it should be `f.Init(...)` instead:
```suggestion
	return f.Init(
```

**Review by matej-g — COMMENTED** [review]
Nice clean-up! Much more readable and clearer :sparkles:.

I have one suspicion on the failing CI, see the comment.

File: test/e2e/e2ethanos/services.go (hunk 2 above)
Comment (bwplotka) [reply test/e2e/e2ethanos/services.go:717]: DID you just save my day? ; p

**Comment by bwplotka** [comment]
image

Go go go!

**Comment by bwplotka** [comment]
PTAL @GiedriusS @matej-g @yeya24

**Review by yeya24 — APPROVED** [review]
LGTM

**Pull request merged**